flexmeasures/data/tests/test_scheduling_repeated_jobs.py

```diff
@@ -0,0 +1,397 @@
+    assert hash == "4ed0V9h247brxusBYk3ug9Cy7miPnynOeSNBT8hd5Mo="
+
+    kwargs2 = copy.deepcopy(kwargs)
+    args2 = copy.deepcopy(args)
+
+    # checks that hashes are consistent within the same runtime calls
+    hash2 = hash_function_arguments(args2, kwargs2)
+    assert hash2 == hash
+
+    # checks that different arguments yield different hashes
+    kwargs2["resolution"] = timedelta(minutes=12)
+    hash3 = hash_function_arguments(args2, kwargs2)
+    assert hash != hash3
+
+
+def test_scheduling_multiple_triggers(
+    caplog, db, app, add_charging_station_assets, setup_test_data
+):
+    caplog.set_level(
+        logging.INFO
+    )  # setting the logging level of the log capture fixture
+
+    duration_until_target = timedelta(hours=2)
+
+    charging_station = Sensor.query.filter(
+        Sensor.name == "Test charging station"
+    ).one_or_none()
+    tz = pytz.timezone("Europe/Amsterdam")
+    start = tz.localize(datetime(2015, 1, 2))
+    end = tz.localize(datetime(2015, 1, 3))
+    resolution = timedelta(minutes=15)
+    target_datetime = start + duration_until_target
+    soc_start = 2.5
+
+    assert (
+        DataSource.query.filter_by(name="FlexMeasures", type="scheduling script")
+        .where()
+        .one_or_none()
+        is None
+    )  # Make sure the scheduler data source isn't there
+
+    # clear logs
+    caplog.clear()
+
+    jobs = []
+
+    # create jobs
+    for target_soc in [1, 1, 4]:
+        soc_targets = [dict(datetime=target_datetime.isoformat(), value=target_soc)]
+
+        job = create_scheduling_job(
+            sensor=charging_station,
+            start=start,
+            end=end,
+            belief_time=start,
+            resolution=resolution,
+            flex_model={"soc-at-start": soc_start, "soc-targets": soc_targets},
+            enqueue=False,
+        )
+
+        # enqueue & run job
+        app.queues["scheduling"].enqueue_job(job)
+        jobs.append(job)
+
+    work_on_rq(app.queues["scheduling"], exc_handler=exception_reporter)
+
+    job1, job2, job3 = jobs
+
+    print(job1.id, job2.id, job3.id)
+
+    # checking that jobs 1 & 2 they have the same job id
+    assert job1.id == job2.id
+
+    # checking that job3 has different id
+    assert job3.id != job1.id
+
+
+def failing_function(*args, **kwargs):
+    raise Exception()
+
+
+def test_allow_trigger_failed_jobs(
+    caplog, db, app, add_charging_station_assets, setup_test_data
+):
+    @job_cache("scheduling")
+    def create_failing_job(
+        arg1: int,
+        kwarg1: int | None = None,
+        kwarg2: int | None = None,
+    ) -> Job:
+        """
+        This function creates and enqueues a failing job.
+        """
+
+        job = Job.create(
+            failing_function,
+            kwargs=dict(kwarg1=kwarg1, kwarg2=kwarg2),
+            connection=app.queues["scheduling"].connection,
+        )
+
+        app.queues["scheduling"].enqueue_job(job)
```

Comment:
Looks like this now attempts to enqueue the job a second time. The first time being in `job_cache`, because the missing `enqueue` kwarg is `True` by default.

Reply:
Good catch! I've changed to rely on the internal enqueuing of the decorator (as in `create_scheduling_job`)